criu/pie/restorer.c

```diff
@@ -202,10 +202,21 @@ static int restore_creds(struct thread_creds_args *args, int procfd,
 	 * Setup supplementary group IDs early.
 	 */
 	if (args->groups) {
-		ret = sys_setgroups(ce->n_groups, args->groups);
-		if (ret) {
-			pr_err("Can't setup supplementary group IDs: %d\n", ret);
-			return -1;
+		// We may be in an unprivileged user namespace where setgroups
+		// is disabled.  If the current list of groups is already what
+		// we want, skip the call to setgroups.
+		unsigned int gids[32];
+		int n = sys_getgroups(32, gids);
```

Comment:
Call getgroups with 0 to get the size of the array, then allocate. Then call getgroups again with the actual size. 
int n = sys_getgroups(0, NULL);
// don't need to error check (might need to justify)
unsigned int* gids = malloc(n * sizeof(unsigned int)); // please brush up on C, Angie
// TODO: check if gids NULL
n = sys_getgroups(n, gids);